ansible/roles/migrate-kubo-c1/tasks/main.yml

```diff
@@ -0,0 +1,131 @@
+      register: c1_snapshots
+      delegate_to: gitcoin-rust-ceramic-1
+
+    - name: Find latest common snapshot
+      ansible.builtin.shell:
+        cmd: |
+          kubo_snaps="{{ kubo_snapshots.stdout_lines | join('\n') }}"
+          c1_snaps="{{ c1_snapshots.stdout_lines | join('\n') }}"
+          echo "$kubo_snaps" | grep -F "$(echo "$c1_snaps" | sed 's/migrationpool\/data-store@//')" | tail -n 1
+      register: common_snapshot
+      failed_when: common_snapshot.rc != 0 or common_snapshot.stdout == ""
+      delegate_to: localhost
+
+    - name: Display latest common snapshot
+      ansible.builtin.debug:
+        var: common_snapshot.stdout
+
+  run_once: true
+
+- name: Create new snapshot on ipfs node
+  ansible.builtin.shell:
+    cmd: zfs snapshot ipfspool/data-store@$(date +%Y%m%d_%H%M%S)
+  register: new_snapshot
+  delegate_to: gitcoin-go-ipfs-1
+
+- name: Send incremental snapshot to c1 node
+  ansible.builtin.shell:
+    cmd: |
+      zfs send -i {{ common_snapshot.stdout }} {{ new_snapshot.stdout }} | \
+      ssh gitcoin-rust-ceramic-1 'zfs receive migrationpool/data-store'
+  delegate_to: gitcoin-go-ipfs-1
+
+- name: Set output file name for diff results
+  ansible.builtin.set_fact:
+    diff_output_file: "/tmp/zfs_diff_output_$(date +%Y%m%d_%H%M%S).txt"
+    modified_block_file: "/tmp/zfs_modified_blockfiles_$(date +%Y%m%d_%H%M%S).txt"
+    files_to_process: "/tmp/zfs_modified_files_to_process_$(date +%Y%m%d_%H%M%S).txt"
+    already_processed: "/tmp/zfs_already_processed.txt"
+
+
+- name: Run zfs diff on the c1 node
+  block:
+    - name: Execute zfs diff
+      ansible.builtin.shell:
+        cmd: |
+          zfs diff -F {{ common_snapshot.stdout | regex_replace('ipfspool', 'migrationpool') }} {{ new_snapshot.stdout | regex_replace('ipfspool', 'migrationpool') }} | tee "{{ diff_output_file }}"
+      args:
+        executable: /bin/bash
+      register: diff_result
+      delegate_to: gitcoin-rust-ceramic-1
+      become: yes
+    
+    - name: Display diff results in real time
+      ansible.builtin.debug:
+        var: diff_result.stdout_lines
+      when: diff_result.stdout_lines | length > 0
+  
+  strategy: free
+
+# run the migration script on the c1 node only on the changed blocks
+
+- name: Generate list of modified block files
+  ansible.builtin.shell:
+    cmd: |
+      awk '/^M\t\/\t/ {print $3}' "{{ diff_output_file }}" | grep '/go_ipfs_datastore/ipfs-data/blocks/' > "{{ modified_block_file }}"
+  args:
+    executable: /bin/bash
+  delegate_to: gitcoin-rust-ceramic-1
+  become: yes
+
+- name: Exclude files already processed
+  ansible.builtin.shell:
+    cmd: |
+      comm -23 <(sort -u "{{ modified_block_file }}") <(sort -u "{{ already_processed }}") > "{{ files_to_process }}"
+  args:
+    executable: /bin/bash
+  delegate_to: gitcoin-rust-ceramic-1
+  become: yes
+
+
+- name: Run migration and update processed files list
+  block:
+
```

Comment:
@smrz2001 can you send me how you ran the script before?  I'm not sure where we should put the output_store_path

Also i think this will handle what you mentioned about possibly needing to rerun the step if it fails, we can rerun on the last modified files list if needed and it will check against the processed files list.  we keep track of all modified files so we can rerun against them if needed